xDrip/Scenes/Entries/EditTraining/EditTrainingInteractor.swift

```diff
@@ -14,20 +14,63 @@ import UIKit
 
 protocol EditTrainingBusinessLogic {
     func doLoad(request: EditTraining.Load.Request)
+    
+    func doCancel(request: EditTraining.Cancel.Request)
+    func doDone(request: EditTraining.Done.Request)
 }
 
 protocol EditTrainingDataStore {
-    
+    var mode: EditTraining.Mode? { get set }
 }
 
 final class EditTrainingInteractor: EditTrainingBusinessLogic, EditTrainingDataStore {
     var presenter: EditTrainingPresentationLogic?
     var router: EditTrainingRoutingLogic?
+    let formattingWorker = EditTrainingFormattingWorker()
```

Comment:
1. Make it private.
2. Extract interface into a separate protocol.
3. Use it as a protocol and initialise it in Interactor's `init()`.
4. `formatting` is not a good name for interactor's worker, because formatting process is Presenter's responsibility.